### DISCUSSION TITLE
How to patch a configmap with new data through kube vela definition ?

### DISCUSSION THREAD

**Discussion by sambeet99** [kubevela/kubevela#6276] [Q&A]
I am running a webservice type component (lets say Master service) on my k8s cluster that has a configmap (master configmap) under its trait section. I am also planning to deploy other webservices (lets say salve services) onto the cluster which should be able to do a patch merge to the master configmap.

So here I am planning to create a custom trait definition which takes data in parameter section and if I attach this custom trait to slave webservices it should be able to do patch the master configmap with new data.

- Basically I have an application which should be able to collect data from other application dynamically. 
- Here I am using a configmap with master application during the start of the k8s cluster and gradually if any slave applications get deployed they should be able to add data to master application configmap.
